portlist/portlist_macos.go

```diff
@@ -184,6 +188,10 @@ func (im *macOSImpl) addProcesses() error {
 			// starting a new process
 			cmd = ""
 			proto = ""
+			pid = 0
+			if p, err := strconv.Atoi(string(val)); err == nil {
```

Comment:
this is an allocation which the rest of this code is trying to avoid. You could use https://pkg.go.dev/go4.org/mem#ParseInt